### PR TITLE
Make multiraft.Transport an input to NewStore.

### PR DESCRIPTION
**Pull request by bdarnell**
Fix some tests where we were calling Store.Start twice.

@spencerkimball @cockroachdb/developers

### CODE REVIEW

**Comment by bdarnell** [comment]
The transport is closely related to the `KVSender` and `Gossip` objects; it seems like it makes sense to attach it to one of them. However, we often test with one or both of those parameters `nil`, but multiraft always requires a transport.

File: server/node.go [hunk 1]
```diff
@@ -116,7 +117,7 @@ func BootstrapCluster(clusterID string, eng engine.Engine) (*client.KV, error) {
 	// Create a KV DB with a local sender.
 	lSender := kv.NewLocalSender()
 	localDB := client.NewKV(kv.NewTxnCoordSender(lSender, clock), nil)
-	s := storage.NewStore(clock, eng, localDB, nil)
+	s := storage.NewStore(clock, eng, localDB, nil, multiraft.NewLocalRPCTransport())
```
Comment (tnachen) [inline server/node.go]: Once we have the real transport impl we should move to that right? Do we ever need to use the local transport?
Just wondering if we need a factory or configuration to drive this.

Comment (bdarnell) [inline server/node.go]: In the case of the BootstrapCluster function, it is always called in the context of a single-node cluster so it's fine to hard-code LocalRPCTransport here (just like we use NewLocalSender two lines up and pass nil as the Gossip parameter to NewStore.

Elsewhere, the real transport depends on a Gossip object, which is nil in many of our tests. We could make a dummy gossip object in all of those tests, which might theoretically give us a more complete test but I think it's probably not worth it.

**Comment by tnachen** [comment]
LGTM

**Pull request merged**